tests/functional/DatadirTest.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\DbWriter\Mssql\FunctionalTests;
+
+use Keboola\Csv\CsvWriter;
+use Keboola\Csv\Exception;
+use Keboola\Csv\InvalidArgumentException;
+use Keboola\DatadirTests\AbstractDatadirTestCase;
+use Keboola\DatadirTests\DatadirTestSpecificationInterface;
+use Keboola\DbWriter\Configuration\ValueObject\MSSQLDatabaseConfig;
+use Keboola\DbWriter\Mssql\TraitTests\CloseSshTunnelsTrait;
+use Keboola\DbWriter\Mssql\TraitTests\PrepareDatabase;
+use Keboola\DbWriter\Writer\MSSQLConnection;
+use Keboola\DbWriter\Writer\MSSQLConnectionFactory;
+use Psr\Log\Test\TestLogger;
+use RuntimeException;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+
+class DatadirTest extends AbstractDatadirTestCase
+{
+    use PrepareDatabase;
+    use CloseSshTunnelsTrait;
+    public MSSQLConnection $connection;
+    protected string $testProjectDir;
+    public ?string $orderResults = null;
+
+    public function __construct(
+        ?string $name = null,
+        array $data = [],
+        string $dataName = '',
+    ) {
+        putenv('SSH_PRIVATE_KEY=' . file_get_contents('/root/.ssh/id_rsa'));
+        putenv('SSH_PUBLIC_KEY=' . file_get_contents('/root/.ssh/id_rsa.pub'));
+        parent::__construct($name, $data, $dataName);
+    }
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->closeSshTunnels();
+        $connectionFactory = new MSSQLConnectionFactory(new TestLogger());
+
+        // prepare database needs master database
+        $databaseConfig = $this->getDatabaseConfig('master');
+        $this->prepareDatabase(
+            $connectionFactory->create($databaseConfig),
+            (string) getenv('DB_DATABASE'),
+        );
+
+        $databaseConfig = $this->getDatabaseConfig();
+        $this->connection = $connectionFactory->create($databaseConfig);
+        $this->orderResults = null;
+        $this->testProjectDir = $this->getTestFileDir() . '/' . $this->dataName();
+
+        // Load setUp.php file - used to init database state
+        $setUpPhpFile = $this->testProjectDir . '/setUp.php';
+        if (file_exists($setUpPhpFile)) {
+            // Get callback from file and check it
+            $initCallback = require $setUpPhpFile;
+            if (!is_callable($initCallback)) {
+                throw new RuntimeException(sprintf('File "%s" must return callback!', $setUpPhpFile));
+            }
+
+            // Invoke callback
+            $initCallback($this);
+        }
+    }
+
+    protected function tearDown(): void
+    {
+        parent::tearDown();
+
+        $setUpPhpFile = $this->testProjectDir . '/tearDown.php';
```

Comment:
Tohle jsem nenašel, že bys někde použil. Plus to jméno proměnné je matoucí :D Ale možná by nebylo od věci přes to zkontrolovat jestli jsou zapsaná data jak očekáváme.

Reply:
jo to by tu asi nemuselo být, vyhodím to...

Reply:
každopádně, tabulky se exportují do složky "db-dump" a pak se to porovnává s tím co je v expected
https://github.com/keboola/db-writer-mssql/pull/78/files#diff-9c18aec8b0cd86ba37b077afe03dc794948e784c44de0ee94e282a87aad813b7R104

Reply:
https://github.com/keboola/db-writer-mssql/pull/78/commits/766b3ff514d1599e0ae6790b49c8cc64a0eaba74